astropy/units/tests/test_format.py

```diff
@@ -358,14 +358,7 @@ def test_deprecated_did_you_mean_units():
     try:
         u.Unit('ANGSTROM', format='fits')
     except ValueError as e:
-        assert 'angstrom (deprecated)' in six.text_type(e)
-        assert 'Angstrom (deprecated)' in six.text_type(e)
-        assert '10**-1 nm' in six.text_type(e)
-
-    with catch_warnings() as w:
-        u.Unit('Angstrom', format='fits')
-    assert len(w) == 1
-    assert '10**-1 nm' in six.text_type(w[0].message)
+        assert 'Did you mean Angstrom or angstrom?' in six.text_type(e)
```

Comment:
ah, I missed that it's not a class attribute. 

However the test is named `test_deprecated_did_you_mean_units` and that purpose would be lost now because you're not testing a "deprecated" unit anymore.

I have to admit I don't know what would be the best approach here. I don't think we should invalidate tests (with `io.fits` these are really important 😄 ) but using dependency injection (or a temporary decorator) or something else here might overcomplicate future changes.